test/Dialect/Comb/canonicalization.mlir

```diff
@@ -1662,3 +1662,60 @@ hw.module @cantCombineOppositeNonBinCmpIntoConstant(in %tag_0: i4, in %tag_1: i4
             %opposite_xor_or, %opposite_xor_and :
             i1, i1, i1, i1, i1, i1, i1, i1, i1, i1, i4, i4
 }
+
+// Ensure canonicalization is not confused by i0.
+// CHECK-LABEL: hw.module @i0checks
+// CHECK-SAME: in %[[I32_VAL:[^ ]*]] : i32, in %[[I1_VAL:[^ ]*]] : i1, in %[[I0_VAL:[^ ]*]] : i0
+// CHECK-DAG: %[[FALSE:.*]] = hw.constant false
+// CHECK-DAG: %[[ZERO:.*]] = hw.constant 0 : i0
+// CHECK-DAG: %[[CONCAT:.*]] = comb.concat %[[I0_VAL]], %[[I32_VAL]] : i0, i32
+// CHECK-DAG: %[[CONCAT_REVERSE:.*]] = comb.concat %[[I32_VAL]], %[[I0_VAL]] : i32, i0
+// CHECK-DAG: %[[CONCAT_REMAINS:.*]] = comb.concat %[[I32_VAL]], %[[I0_VAL]], %i1_val : i32, i0, i1
+// CHECK-DAG: %[[CONCAT_ITSELF:.*]] = comb.concat %[[I0_VAL]], %[[ZERO]] : i0, i0
```

Comment:
Is there a particular reason the `i0` values are not canonicalized away, or is it just not implemented yet?

Reply:
It is not implemented yet. I feel like it would be good to separate fixes from new features.